site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,177 @@
+
+{% include_codefile 'demo/finished/biz-e-corp/src/containers/WorkerContainerContainer.ts' lines:10-12 %}
+
+These final lines define the actual `WorkerContainerContainer` class and export it. The `Container` function creates the class by taking in three parameters - the widget's class definition (alternatively, a widget's registry key can be used), the registry key for the `Injector`, and an object literal that provides the mapping functions used to reconcile the two sets of properties and children that the container can receive (one from the `Injector` and one from the parent widget). The returned class is, in fact, a widget (i.e. it descends from `WidgetBase`) and, therefore, can be used just like any other widget.
+
+The other container that we need is the `WorkerFormContainer`.
+
+{% instruction 'Add the following code to the `WorkerFormContainer` module in the `containers` sub-package.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/containers/WorkerFormContainer.ts' %}
+
+This module is almost identical to the `WorkerContainerContainer` except that additional properties are required by the `WorkerForm` to allow it to react when the user enters information into the form. The `ApplicationContext` contains two methods for managing these events - `onFormInput` and `onFormSave`. These methods need to be passed into the `WorkerForm` to handle the events, but they need to execute in the context of the `ApplicationContext`. To handle this, `bind` is called on each of the methods to explicitly set their execution contexts.
+
+At this point, we have created the `ApplicationContext` to manage state, an `Injector` to inject state into the application's widgets, and `Containers` to manage how properties and children from the injector and parent widgets are combined. In the next section, we will integrate these components into our application.
+
+{% section %}
+
+## Using state containers
+
+{% task 'Integrate containers into an application.' %}
+
+As mentioned in the previous section `Containers` are specialized form of widget since they descend from `WidgetBase`. Given that, they can be used just like any other widget. In our demo application, we can take advantage of that fact by simply replacing the `WorkerForm` and `WorkerContainer` with their container equivalents.
+
+{% instruction 'Replace the imports in the `App` module with the following.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:1-6 %}
+
+There are two major changes to the `App` module's imports. First, the widgets (`WorkerForm` and `WorkerContainer`) have been replaced by their container equivalents (`WorkerFormContainer` and `WorkerContainerContainer`). Second, all of the objects related to state management have been removed, `StatefulMixin`, `WorkerFormData`, and `WorkerProperties`. These are no longer needed since the `App` class no longer needs to manage state.
```

Comment:
"to manage state" -> "to manage _______ state" where _______ is more specific please